src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxWriterExtensions.cs

```diff
@@ -0,0 +1,296 @@
+        {
+            writer.Write(node.Value);
+        }
+
+        public static void WriteFloatValue(this ISyntaxWriter writer, FloatValueNode node)
+        {
+            writer.Write(node.Value);
+        }
+
+        public static void WriteStringValue(this ISyntaxWriter writer, StringValueNode node)
+        {
+            if (node.Block)
+            {
+                writer.Write("\"\"\"");
+
+                var lines = node.Value
+                    .Replace("\"\"\"", "\\\"\"\"")
+                    .Replace("\r", string.Empty)
+                    .Split('\n');
+
+                foreach (var line in lines)
+                {
+                    writer.WriteLine();
+                    writer.WriteIndent();
+                    writer.Write(line);
+                }
+
+                writer.WriteLine();
+                writer.WriteIndent();
+                writer.Write("\"\"\"");
+            }
+            else
+            {
+                writer.Write($"\"{WriteEscapeCharacters(node.Value)}\"");
+            }
+        }
+
+
+        private static string WriteEscapeCharacters(string input)
+        {
+            var stringBuilder = new StringBuilder();
+
+            for (int i = 0; i < input.Length; i++)
+            {
+                char c = input[i];
+                WriteEscapeCharacter(stringBuilder, in c);
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        private static void WriteEscapeCharacter(
+            StringBuilder stringBuilder, in char c)
+        {
+            switch (c)
+            {
+                case '"':
+                    WriteEscapeCharacterHelper(stringBuilder, '"');
+                    break;
+                case '\\':
+                    WriteEscapeCharacterHelper(stringBuilder, '\\');
+                    break;
+                case '/':
+                    WriteEscapeCharacterHelper(stringBuilder, '/');
+                    break;
+                case '\b':
+                    WriteEscapeCharacterHelper(stringBuilder, 'b');
+                    break;
+                case '\f':
+                    WriteEscapeCharacterHelper(stringBuilder, 'f');
+                    break;
+                case '\n':
+                    WriteEscapeCharacterHelper(stringBuilder, 'n');
+                    break;
+                case '\r':
+                    WriteEscapeCharacterHelper(stringBuilder, 'r');
+                    break;
+                case '\t':
+                    WriteEscapeCharacterHelper(stringBuilder, 't');
+                    break;
+                default:
+                    stringBuilder.Append(c);
+                    break;
+            }
+        }
+
+        private static void WriteEscapeCharacterHelper(StringBuilder stringBuilder, in char c)
+        {
+            stringBuilder.Append('\\');
+            stringBuilder.Append(c);
+        }
+
```

Comment:
I think we can spare this instance when we would pass `ISyntaxWriter` along
```suggestion
            else
            {
                writer.Write('"');
                WriteEscapeCharacters(writer, node.Value);
                writer.Write('"'); 
            }
        }


        private static void WriteEscapeCharacters(ISyntaxWriter writer, string input)
        { 
            for (int i = 0; i < input.Length; i++)
            {
                char c = input[i];
                WriteEscapeCharacter(writer, in c);
            } 
        }

        private static void WriteEscapeCharacter(
            ISyntaxWriter writer, in char c)
        {
            switch (c)
            {
                case '"':
                    WriteEscapeCharacterHelper(writer, '"');
                    break;
                case '\\':
                    WriteEscapeCharacterHelper(writer, '\\');
                    break;
                case '/':
                    WriteEscapeCharacterHelper(writer, '/');
                    break;
                case '\b':
                    WriteEscapeCharacterHelper(writer, 'b');
                    break;
                case '\f':
                    WriteEscapeCharacterHelper(writer, 'f');
                    break;
                case '\n':
                    WriteEscapeCharacterHelper(writer, 'n');
                    break;
                case '\r':
                    WriteEscapeCharacterHelper(writer, 'r');
                    break;
                case '\t':
                    WriteEscapeCharacterHelper(writer, 't');
                    break;
                default:
                    writer.Write(c);
                    break;
            }
        }

        private static void WriteEscapeCharacterHelper( ISyntaxWriter writer, in char c)
        {
            stringBuilder.Write('\\');
            stringBuilder.Write(c);
        }

```

Reply:
Or is ` in char c` a problem? 🤔

Reply:
no ... I migrated this code from the old serializer and had also this still on my list. Good job!